cxx-squid/src/main/java/org/sonar/cxx/CxxAstScanner.java

```diff
@@ -209,4 +221,23 @@ public SourceCode createSourceCode(SourceCode parentSourceCode, AstNode astNode)
     return builder.build();
   }
 
+  // Concatenate two strings, but if there is overlap at the intersection,
```

Comment:
For sure ;->

If you are inside of a namespace/class it is allowed to do a definition with the namespace/class again. This function ensures that namespace/class is not doubled. 

 namespace example {
     class A {
          void f1();
          void f2();
          void f3() {} // example::A::f3
          void A::f4() {} // example::A::f4
      };
     void example::A::f1() {} // example::A::f1
     void A::f2() {} // example::A::f2
 }